### PR TITLE
Cryptex: Initial draft

### PR DESCRIPTION
**Pull request by TomFrost**
I need a cover image!

### CODE REVIEW

File: _posts/2016-01-08-learn-this-one-weird-trick-to-protect-your-customers-data.markdown [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+layout: post
+title: "Learn this One Weird Trick to Protect Your Customers' Data"
+date: 2016-01-08 10:00:00
+categories: security
+author: tom.shawver
+cover: /assets/images/covers/books.jpg
+description: With so much focus on secure user passwords recently, there's surprisingly little about secure database and service credentials. Encrypting your credentials against everyone from your CI to your engineers can be simple and quick to implement.
+---
+
+In your web application, how secure do you require yours users' credentials to be? Do you require passwords to be a certain length, with enough kinds of different characters? Maybe you're more progressive and employ two-factor auth, or alternative auth like biometrics or single-use SMS keys. If you do use passwords, maybe you're using a compute-expensive hashing algorithm like bcrypt to keep them safe. Maybe you're encrypting your entire database on disk to keep your users as safe as possible. Nothing short of an attacker compromising your live database could leak that data out.
```
Comment (taghost) [inline _posts/2016-01-08-learn-this-one-weird-trick-to-protect-your-customers-data.markdown]: `require yours users'` to `require your users'`

File: _posts/2016-01-08-learn-this-one-weird-trick-to-protect-your-customers-data.markdown [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+
+If any of those things are true, you're boned. You've locked all the windows and left the front door wide open. If you've not been compromised yet, you simply haven't been a valuable enough [target] yet. When will that happen? Will you know?
+
+## The expensive solution
+
+If you have the budget for launching a server cluster to house and safeguard your secrets, the options there are becoming more plentiful. [Vault] by Hashicorp, Square's [KeyWhiz], and Lyft's [Confidant] have all gotten recent praise, atop an ocean of other capable but less press-generating options.
+
+The security benefits here are fantastic: Not only are your credentials stored encrypted, the central location allows for passwords to be easily rotated. Once your service securely authenticates with one of these server solutions, it gains access to any credentials for which this particular service has been granted access.
+
+The downsides, though, put this out of reach of smaller teams:
```
Comment (taghost) [inline _posts/2016-01-08-learn-this-one-weird-trick-to-protect-your-customers-data.markdown]: `put this out of reach for smaller teams` seems to flow better

**Comment by TomFrost** [comment]
Also up for input: The post title. I was going for ironic and attention-getting, but I'd also like to post this on HN and I'm not sure how I'd feel about that title there. Up for alternatives.

File: _posts/2016-01-08-learn-this-one-weird-trick-to-protect-your-customers-data.markdown [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+  
+## Wrapping up
+
+Cryptex is capable of more than I touched on here, so reading through the [Cryptex Documentation] is encouraged. In particular, if the complexities of Amazon KMS are a turn-off, other key sources are available -- and the consistent API makes it easy to plug other custom ones in.
+
+But if you've gotten this far and I still haven't convinced you how important it is to protect your customers' private data with at least the same level of security you require of their own accounts, let me pose a different question: What do you think your customers would prefer?
```
Comment (taghost) [inline _posts/2016-01-08-learn-this-one-weird-trick-to-protect-your-customers-data.markdown:212]: Should `of their own accounts` be `of your own accounts`?

**Comment by taghost** [comment]
Proofreading complete! Good job! :+1:

File: _posts/2016-01-08-learn-this-one-weird-trick-to-protect-your-customers-data.markdown (hunk 3 above)
Comment (TomFrost) [inline _posts/2016-01-08-learn-this-one-weird-trick-to-protect-your-customers-data.markdown:212]: Nope, of the customers' (their) accounts. I'm tying it back to the opening paragraph that points out how secure we force our customers to be-- burdening them with complex passwords or MFA devices.

**Comment by TomFrost** [comment]
Can I get a final signoff on that last change?

**Comment by ksafranski** [comment]
:goat::dash:

**Pull request merged**